src/contracts/services/derivativeScrt.ts

```diff
@@ -0,0 +1,273 @@
+import { getActiveQueryClient$ } from '~/client';
+import {
+  switchMap,
+  first,
+  map,
+  lastValueFrom,
+} from 'rxjs';
+import { sendSecretClientContractQuery$ } from '~/client/services/clientServices';
+import { convertCoinFromUDenom } from '~/lib/utils';
+import {
+  msgQueryScrtDerivativeFees,
+  msgQueryScrtDerivativeStakingInfo,
+} from '~/contracts/definitions/derivativeScrt';
+import {
+  DerivativeScrtFeeInfoResponse,
+  DerivativeScrtStakingInfoResponse,
+} from '~/types/contracts/derivativeScrt/response';
+import {
+  DerivativeScrtFeeInfo,
+  DerivativeScrtStakingInfo,
+  DerivativeScrtInfo,
+} from '~/types/contracts/derivativeScrt/model';
+import {
+  BatchQueryParsedResponse,
+  BatchQueryParsedResponseItem,
+} from '~/types/contracts/batchQuery/model';
+import { batchQuery$ } from './batchQuery';
+
+// Contract returns price as a rate of stkd-SCRT/SCRT with 6 decimals
+const DERIVATE_PRICE_DECIMALS = 6;
+
+/**
+ * parses the response for the stkd-scrt contract info
+ */
+function parseDerivativeScrtStakingInfo(
+  data: DerivativeScrtStakingInfoResponse,
+): DerivativeScrtStakingInfo {
+  const { staking_info: stakingInfo } = data;
+  const {
+    price: exchangeRate,
+    rewards: communityRewards,
+    total_derivative_token_supply: supply,
+    unbond_amount_of_next_batch: nextUnboundAmount,
+    next_unbonding_batch_time: nextUnbondingBatchEstimatedTime,
+    validators: validatorsResp,
+  } = stakingInfo;
+
+  const validators = validatorsResp.map((nextValidator) => ({
+    validatorAddress: nextValidator.validator,
+    weight: nextValidator.weight,
+  }));
+
+  return {
+    validators,
+    supply: convertCoinFromUDenom(supply, DERIVATE_PRICE_DECIMALS).toString(),
```

Comment:
as a standard in shadejs, we've been returning contract data in raw form and let people do their own conversions elsewhere. I think we can maintain that here as well. Also in our own off chain infra we've been storing in raw, and so this would force us to re-convert back. Let's take the conversions out.